src/scenarios/find-slot.ts

```diff
@@ -34,10 +35,27 @@ export const findSlot = async (wd: WebDriver): Promise<boolean> => {
       wd,
       By.xpath('//*[contains(text(),\'Appointment selection\')]')
     );
-    console.log(`[findSlot]: appointment found`);
-    return true;
+    console.log(`[findSlot]: found calendar`);
+    return !await existsErrorBox(wd)
```

Comment:
so first we try to find 'Appointment selection' meaning the page with calendar appeared, then we try to find a messagesBox (which I think was the initial solution)

Could you elaborate what's the point for waiting for messagesBox after we detected that went to the next page with appointment selection?

PS I might be a little out of the context since I didn't used the script for quite some time

Reply:
we try to find calendar first because if it does, we won't have to wait long. this function searches for an element on page and if the element is not found by timeout it throws an exception.
when we try to find "error box" we wait 60 second and only after that we take exception and try to find calendar.
In this process every second is important =)

Also sometimes, when calendar is found, site writes "we don't have appointment". And it is true. I think that this is bug of this site. because of it I added a check for "error box"

and yes, this functionality is work. I got appointment 4 hours ago

Reply:
maybe you can do it better. I don't know ts =)

Reply:
Okay man thanks for the thorough explanation; I now understand the reasoning behind it. Also, I appreciate your effort in addressing and debugging this issue, it’s not an easy task! With that said, let’s go ahead and merge it!